src/submission/views.py

```diff
@@ -553,33 +553,64 @@ def submit_files(request, article_id):
+                file.delete()
+                messages.add_message(
+                    request,
+                    messages.WARNING,
+                    _('File deleted'),
+                )
+
+                event_logic.Events.raise_event(
+                    event_logic.Events.ON_ARTICLE_FILE_DELETE,
+                    **{
+                        'request': request,
+                        'file_id': file_id,
+                        'original_filename': file.original_filename,
+                        'article': article
+                    }
+                )
+            
             return redirect(reverse('submit_files', kwargs={'article_id': article_id}))
 
         if 'manuscript' in request.POST:
             ms_form = forms.FileDetails(request.POST)
             uploaded_file = request.FILES.get('file')
             if logic.check_file(uploaded_file, request, ms_form):
                 if ms_form.is_valid():
-                    new_file = files.save_file_to_article(
-                        uploaded_file,
-                        article,
-                        request.user,
-                    )
-                    article.manuscript_files.add(new_file)
-                    new_file.label = ms_form.cleaned_data['label']
-                    new_file.description = ms_form.cleaned_data['description']
-                    new_file.save()
-                    return redirect(
-                        reverse('submit_files', kwargs={'article_id': article_id}),
-                    )
+                    try:
+                        with transaction.atomic():
+                            new_file = files.save_file_to_article(
+                                uploaded_file,
+                                article,
+                                request.user,
+                            )
+                            article.manuscript_files.add(new_file)
+                            new_file.label = ms_form.cleaned_data['label']
+                            new_file.description = ms_form.cleaned_data['description']
+                            new_file.save()
+
+                            event_logic.Events.raise_event(
+                                event_logic.Events.ON_ARTICLE_FILE_UPLOAD,
+                                **{
+                                    'request': request,
+                                    'file_id': new_file,
+                                    'original_filename': new_file.original_filename,
+                                    'file_type': 'manuscript',
+                                    'article': article
+                                }
+                            )
+
+                        return redirect(
+                            reverse('submit_files', kwargs={'article_id': article_id}),
+                        )
+                    except Exception as e:
+                        ms_form.add_error(None, str(e))
```

Comment:
A couple of notes here:
1. It is often not very useful to end users to see a system level exception message.
2. The broad `Exception` handler here could swallow errors that should be logged and potentially alert sysadmins.

I think one neat way of solving this would be to pass the form error handler as an optional argument to the event handlers (`ms_form.add_error`) and then tweak the exception handler to something like:

```py
except Exception as e:
    logger.error(e)
    logger.exception(e)
    if not ms_form.errors:
        ms_form.add_error(None,
            f"An unexpected error occurred and has been logged, you can contact support at {support_address}"
        )
```